src/go/rpk/pkg/cli/cluster/connections/list.go

```diff
@@ -0,0 +1,210 @@
+			clauses = append(clauses, fmt.Sprintf("%s = %q", theirs, flag.Value.String()))
+		}
+	}
+
+	idleFlag := fset.Lookup("idle-ms")
+	if idleFlag.Changed {
+		idleMs, err := fset.GetInt64("idle-ms")
+		if err != nil {
+			return nil, err
+		}
+
+		clauses = append(clauses, fmt.Sprintf("idle_duration > %dms", idleMs))
+	}
+	return clauses, nil
+}
+
+func newConnectionList(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Display statistics about current connections",
+		Long:  `Display statistics about current connections.`,
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			prof, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			cl, err := adminapi.NewClient(cmd.Context(), fs, prof)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// Build the filters based on the current flag set
+			filterClauses, err := buildFilters(cmd.Flags())
+			out.MaybeDie(err, "invalid filters: %v", err)
+
+			// TODO: add guardrails and define a proper field mapping
+			orderBy := ""
+			orderByFlag := cmd.Flags().Lookup("order-by")
+			if orderByFlag.Changed {
+				orderBy = orderByFlag.Value.String()
+			}
+
+			// TODO: establish a limit
+			limit, err := cmd.Flags().GetUint32("limit")
+			out.MaybeDie(err, "Invalid limit: %s", err)
+
+			req := adminv2.ListKafkaConnectionsRequest{
+				Filter:   strings.Join(filterClauses, " AND "),
+				OrderBy:  orderBy,
+				PageSize: int32(limit),
+			}
+
+			svc := cl.BrokerService()
+			resp, err := svc.ListKafkaConnections(cmd.Context(), &connect.Request[adminv2.ListKafkaConnectionsRequest]{Msg: &req})
+			out.MaybeDie(err, "error listing connections: %v", err)
+
+			if p.Formatter.IsText() {
+				fmt.Println(printConnectionListTable(resp.Msg))
+			} else {
+				_, _, output, err := p.Formatter.Format(resp.Msg.Connections)
+				out.MaybeDie(err, "error listing connections: %v", err)
+				fmt.Println(output)
+			}
+		},
+	}
+
+	// Set filtering flags
+	fset := cmd.Flags()
+	fset.StringP("state", "s", "", "Filter results by state (OPEN, CLOSED)")
+	fset.String("ip-address", "", "Filter results by the client ip address")
+	fset.String("client-id", "", "Filter results by the client ID")
+	fset.String("client-software-name", "", "Filter results by the client software name")
+	fset.String("client-software-version", "", "Filter results by the client software version")
+	fset.StringP("group-id", "g", "", "Filter by client group ID")
+	fset.Int64P("idle-ms", "i", 0, "Show connections idle for more than i milliseconds")
+	fset.StringP("authentication-user-principal", "u", "", "Filter results by a specific user principal")
+	fset.String("order-by", "", `Order the results by their values. For example, "idle_duration desc" would show connections with the longest idle time first and "total_request_statistics.request_count asc" would show connections with the least requests first`)
```

Comment:
Seems quite long for a flag description, we can add more info about the command in the `Long` field in L152 and redirect users to the `--help flag`, and could also add the field Example and add some examples there

(Similar command: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/cluster/config/list.go#L45)